resin/models/device.py

```diff
@@ -687,3 +690,42 @@ def move(self, uuid, app_name):
             'device', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def get_status(self, uuid):
+        """
+        Get the status of a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        Returns:
+            str: status of a device. List of available statuses: Idle, Configuring, Updating, Offline and Post Provisioning.
+
+        Examples:
+            >>> resin.models.device.get_status('8deb12a58e3b6d3920db1c2b6303d1ff32f23d5ab99781ce1dde6876e8d143')
+            'Offline'
+
+        """
+
+        device = self.get(uuid)
+        print device
+        if device['provisioning_state'] == 'Post-Provisioning':
+            return DeviceStatus.POST_PROVISIONING
+
+        last_seen = datetime.strptime(device['last_seen_time'], "%Y-%m-%dT%H:%M:%S.%fZ")
+        if not device['is_online'] and last_seen.year < self.RESIN_CREATION_YEAR:
+            return DeviceStatus.CONFIGURING
+
+        if not device['is_online']:
+            return DeviceStatus.OFFLINE
+
+        if not device['download_progress'] and device['status'] == 'Downloading':
```

Comment:
Note that with what you have above you'll never get the Downloading state, because you test backwards: you should be testing for 'download_progress' being *set*, not ``not device['download_progress']``